lib/__test__/loader.test.ts

```diff
@@ -47,26 +47,80 @@ describe('hCaptchaLoader', () => {
       });
 
       it('should not fetch script since it was already loaded', async () => {
-          const result = await hCaptchaLoader({ sentry: false });
-          expect(result).toEqual(window.hcaptcha);
-          expect(mockFetchScript).not.toHaveBeenCalled();
+        const result = await hCaptchaLoader({ sentry: false });
+        expect(result).toEqual(window.hcaptcha);
+        expect(mockFetchScript).not.toHaveBeenCalled();
       });
-
     });
 
+    describe('script retry', () => {
+
+      beforeAll(() => {
+        window.hcaptcha = 'hcaptcha-test';
+      })
+
+      afterEach(() => {
+        jest.resetAllMocks();
+        cleanupScripts();
+      });
+
+      it('should retry and load after fetch script error', async () => {
+        mockFetchScript.mockRejectedValueOnce(SCRIPT_ERROR);
+        mockFetchScript.mockResolvedValueOnce(SCRIPT_COMPLETE);
+
+        const promise = hCaptchaLoader({ sentry: false });
+
+        await waitFor(() => {
+          expect(mockFetchScript).toHaveBeenCalledTimes(2);
+
+          // Trigger script onload callback to resolve promise
+          window[HCAPTCHA_LOAD_FN_NAME]();
+          expect(promise).resolves.toEqual(window.hcaptcha);
+        });
+      });
+
+      it('should try loading 2 times and succeed on final try', async () => {
+        mockFetchScript.mockRejectedValueOnce(SCRIPT_ERROR);
```

Comment:
How does this work? Is it queuing these?

Reply:
Correct, they are queued (latest mock is being called first)
If we remove one rejection mock, it will reject only once (`fetchScript` will be called twice) meaning test will fail